ludwig/automl/utils.py

```diff
@@ -178,3 +193,20 @@ def set_output_feature_metric(base_config):
         base_config[HYPEROPT]["metric"] = output_metric
         base_config[HYPEROPT]["goal"] = output_goal
     return base_config
+
+
+def check_imbalanced_output(base_config, features_metadata):
+    """Check binary and category output feature(s) for imbalance, i.e., low minority/majority instance count
+    ratio."""
+    imbalanced_output = False
+    for output_feature in base_config["output_features"]:
+        if output_feature[TYPE] == BINARY or output_feature[TYPE] == CATEGORY:
+            for feature_metadata in features_metadata:
+                if output_feature[NAME] == feature_metadata.name:
+                    if feature_metadata.imbalance_ratio < IMBALANCE_DETECTION_RATIO:
+                        logging.info(
+                            f"Imbalance in {output_feature[NAME]}: minority/majority={feature_metadata.imbalance_ratio}"
+                        )
+                        imbalanced_output = True
+                    break
+    return imbalanced_output
```

Comment:
Would it be more simple to `return True` directly and then `return False` at the end of the function?

Reply:
The function is written so if there is more than one output feature (the break is only out of the inner loop)
and the function returns True if any of the output_features is imbalanced.

Reply:
I see. If there's value in logging each imbalanced output feature, then I agree that we shouldn't short-circuit.